giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/delete/impl/DeleteDocumentService.java

```diff
@@ -0,0 +1,155 @@
+import edu.asu.diging.gilesecosystem.web.core.files.IFilesManager;
+import edu.asu.diging.gilesecosystem.web.core.model.IDocument;
+import edu.asu.diging.gilesecosystem.web.core.model.IFile;
+import edu.asu.diging.gilesecosystem.web.core.service.IFileTypeHandler;
+import edu.asu.diging.gilesecosystem.web.core.service.core.ITransactionalDocumentService;
+import edu.asu.diging.gilesecosystem.web.core.service.core.ITransactionalFileService;
+import edu.asu.diging.gilesecosystem.web.core.service.core.ITransactionalProcessingRequestService;
+import edu.asu.diging.gilesecosystem.web.core.service.core.ITransactionalUploadService;
+import edu.asu.diging.gilesecosystem.web.core.service.delete.IDeleteDocumentService;
+import edu.asu.diging.gilesecosystem.web.core.service.properties.Properties;
+
+/**
+ * This class handles the deletion of a document.
+*/
+
+@Service
+public class DeleteDocumentService implements IDeleteDocumentService {
+    
+    public final static String REQUEST_PREFIX = "DELREQ";
+    
+    @Autowired
+    private IRequestFactory<IStorageDeletionRequest, StorageDeletionRequest> requestFactory;
+    
+    @Autowired
+    private ApplicationContext ctx;
+    
+    @Autowired
+    private ISystemMessageHandler messageHandler;
+    
+    @Autowired
+    private IRequestProducer requestProducer;
+    
+    @Autowired
+    private IPropertiesManager propertyManager;
+    
+    @Autowired
+    private ITransactionalFileService fileService;
+    
+    @Autowired
+    private ITransactionalUploadService uploadService;
+    
+    @Autowired
+    private ITransactionalDocumentService documentService;
+    
+    @Autowired
+    private ITransactionalProcessingRequestService processingRequestService;
+    
+    private Map<String, FileType> fileTypes;
+    
+    @PostConstruct
+    public void init() {
+        fileTypes = new HashMap<String, FileType>();
+        
+        // register what content types should be treated as what file type
+        Map<String, IFileTypeHandler> ctxMap = ctx.getBeansOfType(IFileTypeHandler.class);
+        Iterator<Entry<String, IFileTypeHandler>> iter = ctxMap.entrySet().iterator();
+        
+        while(iter.hasNext()){
+            Entry<String, IFileTypeHandler> handlerEntry = iter.next();
+            IFileTypeHandler handler = (IFileTypeHandler) handlerEntry.getValue();
+            for (String type : handler.getHandledFileTypes()) {
+                fileTypes.put(type, handler.getHandledFileType());
+            }
+        }
+        requestFactory.config(StorageDeletionRequest.class);
+    }
+    
+    @Override
+    @Async
+    public void deleteDocument(IDocument document) {
+        try {
+            IRequest storageDeletionRequest = createRequest(document);
+            requestProducer.sendRequest(storageDeletionRequest, getTopic());
+        } catch (GilesProcessingException | MessageCreationException e) {
+            messageHandler.handleMessage("Could not create Request", e, MessageType.ERROR);
```

Comment:
I think this needs to throw an exception. Otherwise, how would the user know that the deletion failed?